packages/block-library/src/query/utils.js

```diff
@@ -260,3 +261,61 @@ export function useBlockNameForPatterns( clientId, attributes ) {
 	);
 	return activeVariationPatterns?.length ? blockName : queryLoopName;
 }
+
+/**
+ * Helper hook that determines if there is an active variation of the block
+ * and if there are available specific scoped `block` variations connected with
+ * this variation.
+ *
+ * If there are, these variations are going to be the only ones suggested
+ * to the user in setup flow when clicking to `start blank`, without including
+ * the default ones for Query Loop.
```

Comment:
I'm not sure If I got this right, but I understood the same with @gigitux 
>If I understand, @mcsf describes a similar use case I shared with https://github.com/WordPress/gutenberg/pull/46369. It may be a rare use case, and the developer can fix it easily by registering the same variation for both namespaces.

Also my [comment](https://github.com/WordPress/gutenberg/pull/46369#issuecomment-1344247870) there: 
>Hm.. While it's possible, it seems a bit weird to me to have variations with scope both inserter and block. Personally I think that scoped block variations in Query Loop(and the other core blocks) are there to provide starting innner blocks and not suggest an entirely different variation like the products on sale etc.. It doesn't seem to fit the expected flow, that is start blank, but 🤷 .

Maybe related to that, I also [added](https://github.com/WordPress/gutenberg/pull/46410/commits/fb0fe82790bf861909729f59492aef7c2b6cd49a) multiple connected variations support, by making the `namespace` an array of names.